src/ground-truth.rkt

```diff
@@ -33,12 +33,12 @@
 ;; Returns a function that maps an ival to a list of ivals
 ;; The first element of that function's output tells you if the input is good
 ;; The other elements of that function's output tell you the output values
-(define (make-search-func precondition programs ctx)
-  (define fns (batch-eval-progs (cons precondition programs) 'ival ctx))
+(define (make-search-func precondition programs ctxlist)
+  (define fns (batch-eval-progs (cons precondition programs) 'ival (car ctxlist)))
   (λ inputs
-    (define repr (context-repr ctx))
     (match-define (list ival-pre ival-bodies ...) (apply fns inputs))
     (for/list ([y ival-bodies])
+      (define repr (context-repr (car ctxlist)))
```

Comment:
Uhh, is this right? I'd expect that you need to iterate through the `ctxlist` in parallel with the `ival-bodies`. Since:
- `ival-bodies` came out of `fns`
- `fns` produces output parallel to `programs`
- `ctxlist` is parallel to `programs`

Reply:
Think I made this change correctly. Is there a test we should add to protect against this potential bug?